ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,389 @@
+        )
+        self._mc_header = self._parse_mc_header(
+            self.file_.root.configuration.simulation.run
+        )
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.file_.close()
+
+    @staticmethod
+    def is_compatible(file_path):
+        """
+        Implementation needed!
+        """
+        return False
+
+    @property
+    def is_simulation(self):
+        """
+        Implementation needed!
+        """
+        return True
+
+    @property
+    def subarray(self):
+        return self._subarray_info
+
+    @property
+    def datalevels(self):
+        return (DataLevel.DL1_IMAGES, DataLevel.DL1_PARAMETERS)
+
+    @property
+    def obs_id(self):
+        return set(self.file_.root.dl1.event.subarray.trigger.col("obs_id"))
+
+    @property
+    def mc_header(self):
+        return self._mc_header
+
+    def _generator(self):
+        try:
+            yield from self._generate_events()
+        except EOFError:
+            msg = 'EOFError reading from "{input_url}". Might be truncated'.format(
+                input_url=self.input_url
+            )
+            self.log.warning(msg)
+            warnings.warn(msg)
+
+    def _prepare_subarray_info(self, instrument_description):
+        """
+        Constructs a SubArrayDescription object from
+        self.file_.root.configuration.instrument.telescope.optics
+        and
+        self.file_.root.configuration.instrument.subarray.layout
+        tables.
+
+        Returns
+        -------
+        SubarrayDescription :
+            instrumental information
+        """
+        available_optics = instrument_description.telescope.optics.iterrows()
+        available_telescopes = instrument_description.subarray.layout.iterrows()
+
+        # The focal length choice is missing here
+        # I am not sure how they are handled in the file
+        # Will there be one of "e_fl" or "fl" in the columns?
+        # This will only work if "e_fl" is available
+        optic_descriptions = {}
+        for optic in available_optics:
+            optic_description = OpticsDescription(
+                name=optic['name'].decode(),
+                num_mirrors=optic['num_mirrors'],
+                equivalent_focal_length=u.Quantity(
+                    optic['equivalent_focal_length'],
+                    u.m,
+                ),
+                mirror_area=u.Quantity(
+                    optic['mirror_area'],
+                    u.m ** 2,
+                ),
+                num_mirror_tiles=optic['num_mirror_tiles'],
+            )
+            optic_descriptions[optic['description'].decode()] = optic_description
+
+        tel_positions = {}
+        tel_descriptions = {}
+        for telescope in available_telescopes:
+            tel_positions[telescope['tel_id']] = (
+                telescope['pos_x'],
+                telescope['pos_y'],
+                telescope['pos_z'],
```

Comment:
the positions would need an unit too
```
np.array([telescope['pos_x'],
                telescope['pos_y'],
                telescope['pos_z']]) * available_telescopes['pos_x'].unit
```

Reply:
Good catches, but I don't think there is much point in reviewing the code right now. There is a lot, that is going to change once the container IO works the way we want it to.